base/src/main/java/edu/artic/base/ObservableExtensions.kt

```diff
@@ -41,4 +43,20 @@ fun <T> Observable<Result<T>>.mapWithDefault(onErrorValue : T) : Observable<T> {
 fun <T> Observable<T>.waitForASecondOfCalmIn(other: Subject<*>): Observable<T> {
     return zipWith(other.debounce(1, TimeUnit.SECONDS))
             .map { (original, _) -> original }
-}
\ No newline at end of file
+}
+
+/**
+ * Extension method for handling network errors.
+ * TODO:: Use localized the error message.
+ */
+fun <T> Observable<T>.handleNetworkError(): Observable<T> {
```

Comment:
Can we use string resources for the `NetworkException` messages? Passing a context or resources in as parameter might work.

Reply:
Updated. Passes the context.